functional/tools/stresstest/stresstest.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/spiffe/spire/proto/api/registration"
+	"github.com/spiffe/spire/proto/common"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	serverAddr           = "localhost:8081"
+	parentSpiffeIDPrefix = "spiffe://example.org/spire/agent/join_token/"
+	spiffeIDPrefix       = "spiffe://example.org/"
+	workloadPath         = "/go/src/github.com/spiffe/spire/functional/tools/workload/workload"
+)
+
+type workloadStats struct {
+	runtime time.Duration
+	output  string
+	success bool
+}
+
+func main() {
+	token := flag.String("token", "", "Join token used in server and agent")
+	users := flag.Int("workloads", 5, "Number of workloads to run in parallel")
+	timeout := flag.Int("timeout", 15, "Total time to run test")
+	ttl := flag.Int("ttl", 120, "SVID TTL")
+	flag.Parse()
+
+	if *token == "" {
+		flag.Usage()
+		return
+	}
+
+	c, err := newRegistrationClient(serverAddr)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create users and register workloads
+	for i := 0; i < *users; i++ {
+		uid := 1000 + i
+
+		fmt.Printf("Creating user %d\n", uid)
+
+		// Create user
+		o, err := exec.Command("bash", "-c", fmt.Sprintf("useradd --uid %d user%d", uid, uid)).CombinedOutput() //adduser --uid 1111 --disabled-password --shell /bin/bash --ingroup spire spire
+		if err != nil {
+			fmt.Println(string(o))
+			panic(err)
+		}
+
+		// Register workload
+		parentID := parentSpiffeIDPrefix + *token
+		selectorValue := fmt.Sprintf("uid:%d", uid)
+		spiffeID := spiffeIDPrefix + fmt.Sprintf("uid%d", uid)
+		fmt.Printf("Parent ID: %s\nSelector Value: %s\nSpiffe ID: %s\n", parentID, selectorValue, spiffeID)
+		entry := &common.RegistrationEntry{
+			ParentId: parentID,
+			Selectors: []*common.Selector{
+				&common.Selector{
+					Type:  "unix",
+					Value: selectorValue,
+				},
+			},
+			SpiffeId: spiffeID,
+			Ttl:      int32(*ttl),
+		}
+		entryID, err := c.CreateEntry(context.TODO(), entry)
+		if err != nil {
+			panic(err)
+		}
+		fmt.Printf("Created entry ID %s\n", entryID.Id)
+	}
+
+	fmt.Printf("Waiting for entries to be propagated...\n")
+	time.Sleep(time.Second * time.Duration(10))
+
+	stats := make(map[int]*workloadStats)
+
+	var wg sync.WaitGroup
+	wg.Add(*users)
```

Comment:
Generally speaking, it's good to call Add as close to the goroutine declaration as possible... Helps to keep the group accurate if we encounter a failure part way through, and is a little easier to read.